src/store/search.js

```diff
@@ -400,7 +398,7 @@ const actions = {
     const newFilterData = queryToFilterData({
       query,
       searchType,
-      defaultFilters: state.filters,
+      defaultFilters: filterData,
```

Comment:
This is the actual fix. The rest is all support/testing/validation.

Reply:
We need to add provider filters to this object, because originally filterData doesn't have it, and it is downloaded in nuxtServerInit. Actually, since the number of providers doesn't change that often we could hard-code it, and update it whenever a new provider is added.